### PR TITLE
Add `Enhance` types to the session docs

### PR DESCRIPTION
**Pull request by vrumger**

### CODE REVIEW

**Comment by Loskir** [comment]
It is missing the point: that you have to use Enhance inside SessionFlavor in your Context

`type MyContext = Context & SessionFlavor<Enhance<SessionData>>`

**Comment by vrumger** [comment]
@Loskir that causes typescript to complain that I need `session.__d.whatever`.

File: site/docs/plugins/session.md [hunk 1]
```diff
@@ -694,6 +694,24 @@ The session plugin is able to enhance any storage adapter by adding more feature
 
 They can be installed using the `enhanceStorage` function.
 
+<CodeGroup>
+<CodeGroupItem title="TypeScript" active>
+
+```ts
+// Create a storage adapter.
+const storage = freeStorage<Enhance<SessionData>>(bot.token); // adjust this
```
Comment (Loskir) [inline site/docs/plugins/session.md:702]: at the same time this is wrong. you don't need Enhance here

Comment (vrumger) [reply site/docs/plugins/session.md:702]: that was the only way I can get it to work properly.

Comment (Loskir) [reply site/docs/plugins/session.md:702]: or do you?..

**Review by KnorpelSenf — APPROVED** [review]
Thank you so much! This is great, it should have been written much earlier.

**Comment by Loskir** [comment]
Sorry, I was wrong. You don't need to use `Enhance` at all, it seems.

At least if you write it this way:

```ts
interface SessionData {
  n: number
}

type MyContext = Context & SessionFlavor<SessionData>

const bot = new Bot<MyContext>('')

bot.use(session({
  storage: enhanceStorage({
    storage: new MemorySessionStorage()
  })
}))
```

What do you think about changing the code in the docs to this?

**Comment by KnorpelSenf** [comment]
Can you check if it works with at least one other storage adapter as well as the free sessions?

**Comment by KnorpelSenf** [comment]
Well actually it makes sense now that I think about it, because we didn't give TS a way to infer the types before and now we do 🤔

File: site/docs/plugins/session.md [hunk 2]
```diff
@@ -694,6 +694,24 @@ The session plugin is able to enhance any storage adapter by adding more feature
...
+  // more config here
+});
+// Use the enhanced storage adapter.
+bot.use(session({ storage: enhanced }));
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="JavaScript">
+
 ```ts
```
Comment (CikiMomogi) [inline site/docs/plugins/session.md:715]:
````suggestion
```js
````

File: site/docs/id/plugins/session.md [hunk 3]
```diff
@@ -700,6 +700,24 @@ Plugin session dapat meningkatkan kemampuan storage adapter dengan cara menambah
...
+  // tulis konfigurasinya di sini
+});
+// Gunakan storage adapter yang sudah ditingkatkan.
+bot.use(session({ storage: enhanced }));
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="JavaScript">
+
 ```ts
```
Comment (CikiMomogi) [inline site/docs/id/plugins/session.md:721]:
````suggestion
```js
````

**Comment by vrumger** [comment]
> Can you check if it works with at least one other storage adapter as well as the free sessions?

@KnorpelSenf yep, it works

**Comment by KnorpelSenf** [comment]
Nice, then please go ahead with that change! I agree it's much more readable than fixing up the broken thing that I wrote

**Review by roj1512 — APPROVED** [review]

File: site/docs/id/plugins/session.md [hunk 4]
```diff
@@ -700,16 +700,14 @@ Plugin session dapat meningkatkan kemampuan storage adapter dengan cara menambah
...
 
-```ts
-// Buat sebuah storage adapter.
-const storage = freeStorage(bot.token); // jangan lupa diatur
-// Tingkatkan kemampuan storage.
-const enhanced = enhanceStorage({
-  storage,
-  // tulis konfigurasinya di sini
-});
+```js
```
Comment (Loskir) [inline site/docs/id/plugins/session.md:703]:
```suggestion
```ts
```

**Comment by CikiMomogi** [comment]
We should merge it to a local branch to process the translations.

**Comment by CikiMomogi** [comment]
Btw, which section has to be translated? :thinking:

**Comment by Loskir** [comment]
Indonesian part is done, it seems

**Comment by CikiMomogi** [comment]
@KnorpelSenf @roj1512  It seems the changes from https://github.com/grammyjs/website/pull/455 not synced in zh and es. I was going to help translate them, but the section is not available in their documentation.

**Pull request merged**